crates/holochain/src/conductor/conductor/tests/agent_key_revocation.rs

```diff
@@ -0,0 +1,199 @@
+    assert!(result.is_some());
+
+    // creating an entry should fail now for both cells
+    let result = conductor
+        .call_fallible::<_, ActionHash>(&zome_1, create_fn_name, ())
+        .await;
+    if let Err(ConductorApiError::CellError(CellError::WorkflowError(workflow_error))) = result {
+        assert_matches!(
+            *workflow_error,
+            WorkflowError::SysValidationError(SysValidationError::ValidationOutcome(ValidationOutcome::DpkiAgentInvalid(invalid_key, _))) if invalid_key == agent_key
+        );
+    } else {
+        panic!("different error than expected");
+    }
+    let result = conductor
+        .call_fallible::<_, ActionHash>(&zome_2, create_fn_name, ())
+        .await;
+    if let Err(ConductorApiError::CellError(CellError::WorkflowError(workflow_error))) = result {
+        assert_matches!(
+            *workflow_error,
+            WorkflowError::SysValidationError(SysValidationError::ValidationOutcome(ValidationOutcome::DpkiAgentInvalid(invalid_key, _))) if invalid_key == agent_key
+        );
+    } else {
+        panic!("different error than expected");
+    }
+
+    // last source chain action in both cells should be CloseChain
+    let sql = "SELECT type FROM Action ORDER BY seq DESC LIMIT 1";
+    let row_fn = |row: &Row| {
+        let action_type = row.get::<_, String>("type").unwrap();
+        assert_eq!(action_type, ActionType::CloseChain.to_string());
+        Ok(())
+    };
+    conductor
+        .get_or_create_authored_db(dna_file_1.dna_hash(), agent_key.clone())
+        .unwrap()
+        .test_read(move |txn| {
+            txn.query_row(sql, [], row_fn.clone()).unwrap();
+        });
+    conductor
+        .get_or_create_authored_db(dna_file_2.dna_hash(), agent_key.clone())
+        .unwrap()
+        .test_read(move |txn| {
+            txn.query_row(sql, [], row_fn).unwrap();
+        });
+
+    // cloning cells should fail for both cells
+    let mut create_clone_cell_payload = CreateCloneCellPayload {
+        role_name: role_1.to_string(),
+        membrane_proof: None,
+        modifiers: DnaModifiersOpt::none().with_network_seed("network_seed".into()),
+        name: None,
+    };
+    let result = conductor
+        .raw_handle()
+        .create_clone_cell(app.installed_app_id(), create_clone_cell_payload.clone())
+        .await
+        .unwrap_err();
+    assert_matches!(result, ConductorError::AppError(AppError::CellToCloneHasInvalidAgent(invalid_key)) if invalid_key == agent_key);
+    create_clone_cell_payload.role_name = role_2.to_string();
+    let result = conductor
+        .raw_handle()
+        .create_clone_cell(app.installed_app_id(), create_clone_cell_payload)
+        .await
+        .unwrap_err();
+    assert_matches!(result, ConductorError::AppError(AppError::CellToCloneHasInvalidAgent(invalid_key)) if invalid_key == agent_key);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn delete_agent_key_without_dpki_installed_fails() {
```

Comment:
```suggestion
async fn revoke_agent_key_without_dpki_installed_fails() {
```

I think only revocation should fail because the key isn't managed. At some point, I see no reason why deleting an agent key should fail. It's fine that we don't implement it now but it should be possible to remove a standard agent key from Lair and get into the same state as you would if you revoked a key.